src/main/java/org/fibsters/FibSpiralComputeEngineImpl.java

```diff
@@ -97,12 +183,73 @@ public void setOutputPayload(OutputPayloadImpl outputPayload) {
 
     @Override
     public void run() {
+
+
         this.status = ComputeJobStatus.RUNNING;
+        this.generateValues();
+
+        draw();
+
+        this.status = ComputeJobStatus.COMPLETED;
+    }
+
+    private void draw() {
         BufferedImage image = this.outputPayload.getOutputImage(); //image to write to
         //TODO: do the writing that's in the legacy class
 
+        Graphics2D graphics = (Graphics2D) image.getGraphics();
+        graphics.setColor(Color.BLUE);
 
-        this.status = ComputeJobStatus.COMPLETED;
+        synchronized (image) {
+            // draw a dot at the center of the image
+            double arcX = posX;
+            double arcY = posY;
+
+            if (angle == 0) {
+                arcX = posX - scaledFib;
+                arcY = posY;
+                graphics.setColor(Color.GRAY);
+            } else if (angle == 90) {
+                arcX = posX;
+                arcY = posY;
+                graphics.setColor(Color.WHITE);
+            } else if (angle == 180) {
+                arcX = posX;
+                arcY = posY - scaledFib;
+                graphics.setColor(Color.ORANGE);
+            } else if (angle == 270) {
+                arcX = posX - scaledFib;
+                arcY = posY - scaledFib;
+                graphics.setColor(Color.GREEN);
+            }
+
+            Shape shape = new BigIntRectangle(BigIntUtil.toBigInt(posX), BigIntUtil.toBigInt(posY), BigIntUtil.toBigInt(scaledFib), BigIntUtil.toBigInt(scaledFib));
+
+            graphics.draw(shape);
+            graphics.setColor(Color.RED);
+
+            Arc2D arc = new Arc2D.Double(arcX, arcY, 2 * scaledFib - 1, 2 * scaledFib - 1, angle, 90, Arc2D.OPEN);
+
+            graphics.draw(arc);
+
+            graphics.drawString(Double.toString(currentFib), (int) (posX + scaledFib / 2), (int) (posY + scaledFib / 2));
+        }
+
+        graphics.dispose();
+
+        String fileName = this.outputPayload.getFileLocations()[this.chunk];
+
+        saveImage(image, fileName);
+    }
+
+    private void saveImage(BufferedImage image, String fileName) {
```

Comment:
nice work on saving the final image here, allows for user output of files to user-specified format